### PR TITLE
drivers: spi: Change to use software controlled cs in default

### PR DESCRIPTION
**Pull request by AlessandroLuo**
Added support for software controlled cs in Ambiq SPI drivers

### CODE REVIEW

**Review by aaronyegx — CHANGES_REQUESTED** [review]
Break the HCI driver, hold on this

**Review by RichardSWheatley — DISMISSED** [review]

**Review by tbursztyka — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by aaronyegx — DISMISSED** [review]
The Ambiq HCI driver and BLE samples are back to normal now.

**Review by aaronyegx — APPROVED** [review]

**Comment by jhedberg** [comment]
> The Ambiq HCI driver and BLE samples are back to normal now.

Does that mean that without this PR the driver is broken? In that case it'd make sense to reconsider changing the milestone from 4.0 (which @nashif set) to 3.7.

**Comment by jhedberg** [comment]
> > The Ambiq HCI driver and BLE samples are back to normal now.
> 
> Does that mean that without this PR the driver is broken? In that case it'd make sense to reconsider changing the milestone from 4.0 (which @nashif set) to 3.7.

Ah, I see it was probably a reference to https://github.com/zephyrproject-rtos/zephyr/pull/75064#pullrequestreview-2144885567 i.e. the driver was only broken for an earlier iteration of this PR but works fine with current main branch?

**Comment by AlessandroLuo** [comment]
> > > The Ambiq HCI driver and BLE samples are back to normal now.
> > 
> > 
> > Does that mean that without this PR the driver is broken? In that case it'd make sense to reconsider changing the milestone from 4.0 (which @nashif set) to 3.7.
> 
> Ah, I see it was probably a reference to [#75064 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/75064#pullrequestreview-2144885567) i.e. the driver was only broken for an earlier iteration of this PR but works fine with current main branch?

Correct.

File: boards/ambiq/apollo3_evb/apollo3_evb.dts [hunk 1]
```diff
@@ -106,6 +106,7 @@
 	compatible = "ambiq,spi";
 	pinctrl-0 = <&spi0_default>;
 	pinctrl-names = "default";
+	cs-gpios = <&gpio0_31 11 GPIO_ACTIVE_LOW>;
```
Comment (RichardSWheatley) [inline boards/ambiq/apollo3_evb/apollo3_evb.dts:109]: I like this much better since we need HOLD_ON_CS.

**Review by RichardSWheatley — APPROVED** [review]
Much better!

**Review by jhedberg — APPROVED** [review]

**Pull request merged**